internal/pkg/agent/cmd/run.go

```diff
@@ -208,7 +177,7 @@ func runElasticAgent(ctx context.Context, cancel context.CancelFunc, override cf
 	}
 
 	// agent ID needs to stay empty in bootstrap mode
-	createAgentID := true
+	createAgentID := !runAsOtel
```

Comment:
Just double checking: this change is not strictly related to this PR but a fix for a small bug you found while working on this PR?

Reply:
yes, on mac it fails to run otel because agent tries to create agent id and do not have permissions if not running as root